beater/beater.go

```diff
@@ -130,16 +132,23 @@ type beater struct {
 // Run runs the APM Server, blocking until the beater's Stop method is called,
 // or a fatal error occurs.
 func (bt *beater) Run(b *beat.Beat) error {
-	tracer, tracerServer, err := initTracer(b.Info, bt.config, bt.logger)
+
+	tracerServer, err := newTracerServer(bt.config, b.Instrumentation.Listener())
```

Comment:
Shouldn't we only be creating a tracerServer if `b.Instrumentation.Listener()` returns non-nil? i.e. when instrumentation is enabled with no hosts specified